lib/nrf_modem_lib/nrf_modem_os.c

```diff
@@ -607,5 +635,24 @@ int32_t nrf_modem_os_trace_put(const uint8_t * const data, uint32_t len)
 	}
 #endif
 
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
+	/* First, let's check if the buffer has been correctly
+	 * allocated for the modem trace
+	 */
+	if (trace_rtt_channel < 0) {
+		return 0;
+	}
+
+	uint32_t remaining_bytes = len;
+
+	while (remaining_bytes) {
+		uint8_t transfer_len = MIN(remaining_bytes, RTT_BUF_SZ);
+		uint32_t idx = len - remaining_bytes;
+
+		SEGGER_RTT_WriteSkipNoLock(trace_rtt_channel, &data[idx],
```

Comment:
@GiulianoFranchetto   If its not too much of a trouble, could you tell me the JLink speed you use (had used) to extract the modem traces via RTT interface? I am testing this feature now and even if I use speeds as high as 16Mhz, this API still fails quite often even to write 255 bytes (the default RTT buffer size you have chosen). This leads to several broken traces. Since the return value of this API is ignored here, this may not have been noticed until now.

Reply:
Hi.
I'm usually using 4MHz, and I have not tested with faster frequency.
However, I don't see why this function should fail on higher bus frequency.
Sounds more like the HAL is not capable of such frequency, but honnestly, I'm not an expert **at all** on RTT protocol.
Maybe @lemrey would have more insight

Reply:
Thanks for replying. The api seems to fail at all speeds. The only way to improve this is to set a higher value ( more than 3000 bytes) to the the RTT buffer size. Right now the default value is 255 bytes.